src/components/NewCommentForm.tsx

```diff
@@ -2,7 +1,73 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { createComment } from '../api/comments';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  postId: number | undefined;
+  comments: Comment[];
+  setComments: (comments: Comment[]) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  postId,
+  comments,
+  setComments,
+}) => {
+  const [authorName, setAuthorName] = useState<string>('');
+  const [authorEmail, setAuthorEmail] = useState<string>('');
+  const [commentBody, setCommentBody] = useState<string>('');
+
+  const [errorAuthorName, setErrorAuthorName] = useState<boolean>(false);
+  const [errorAuthorEmail, setErrorAuthorEmail] = useState<boolean>(false);
+  const [errorCommentBody, setErrorCommentBody] = useState<boolean>(false);
+
+  const [loadingNewComment, setLoadingNewComment] = useState<boolean>(false);
+
+  function resetForm() {
+    setAuthorName('');
+    setAuthorEmail('');
+    setCommentBody('');
+    setErrorAuthorName(false);
+    setErrorAuthorEmail(false);
+    setErrorCommentBody(false);
+  }
+
+  function addComment() {
+    setErrorAuthorName(!authorName);
+    setErrorAuthorEmail(!authorEmail);
+    setErrorCommentBody(!commentBody);
```

Comment:
Consider adding a regex check to validate the email format before setting `errorAuthorEmail`. This will ensure that the email input is valid before allowing the form to be submitted.